### DISCUSSION TITLE
Let's talk about Lodash

### DISCUSSION THREAD

**Discussion by 20lives** [tailwindlabs/tailwindcss#2610] [Ideas]
The Lodash utility library is used widely in this project, and I wanted to discuss it's relevancy today.
When you are targeting old browsers Lodash could be very useful but today many of the library utilities are available natively.
the lowest requirement now is Node v12.13 which supports all ES2019 features (and we are also using babel)
In addition, Lodash implementation makes it  harder to read, understand and reuse logic (Unlike other libraries like Ramda)
And after my minor refactor (a2337e6cc85ab9e97a22ab1389f3ed911255248f)  I think many of the acrobatics used with it might be unnecessary and could be simplified.

I think you should consider dropping it and save 1.4MB for each installation worldwide.

**Comment by adamwathan** [comment]
Happy to accept PRs that replace use of lodash with native JS features now that we dropped node 8 support, but it’ll be a big project to replace entirely, especially some of the more complex features we use like mergeWith. Can’t justify spending my own time on it any time soon as it’s a very low value project compared to the other things on my plate.

**Comment by 20lives** [comment]
I agree with you that this isn't as rewarding as other tasks, but probably only for the short term.
This doesn't have to all in one stroke but incremental work and I will gladly help.

**Comment by tehpsalmist** [comment]
If you find a few lodash functions are useful/hard to replace, you can install and import them separately instead of importing the whole library. That might save you the MBs right there, and is easily done.
